### PR TITLE
Update THStack documentation

### PR DESCRIPTION
**Pull request by couet**
As said in this old forum post https://root-forum.cern.ch/t/thstack-th2-with-negative-values/16161/5, stacking a histogram with negative bin content does not make sense. This issue is not relevant https://github.com/root-project/root/issues/15740. But it is goo to make it clear in the documentation and to prevent adding such histograms in a stack. That's what this PR is doing.

### CODE REVIEW

**Comment by linev** [comment]
>  Therefore, histograms with a negative minimum will be rejected.

How it is done? Currently I see no protection for negative bins content.

**Comment by couet** [comment]
> > Therefore, histograms with a negative minimum will be rejected.
> 
> How it is done? Currently I see no protection for negative bins content.

In Add:

```
if (h1->GetMinimum() < 0.) {
       Error("Add","Histograms with a negative minimum cannot be part of a THStack");
       return;
}
```

**Comment by couet** [comment]
Any way this PR make several test fail like "tutorial-legacy-mlp-mlpRegression" let see what @moneta think about it.

**Comment by couet** [comment]
It seems OK now

File: hist/hist/src/THStack.cxx [hunk 1]
```diff
@@ -400,15 +401,17 @@ void THStack::BuildStack()
    fStack->Add(h);
    for (Int_t i=1;i<nhists;i++) {
       h = (TH1*)fHists->At(i)->Clone();
+      if (h->GetMinimum() < 0.) {
+         Warning("Add","Histograms with a negative minimum may produce wrong plots");
+      }
```
Comment (linev) [inline hist/hist/src/THStack.cxx:406]: Here must be `Warning("BuildStack", ...)`

**Review by linev — APPROVED** [review]
One should change Warning method name.

**Pull request merged**